src/main/java/net/fabricmc/loader/impl/util/mappings/MixinRemapper.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright 2016 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.loader.impl.util.mappings;
+
+import org.spongepowered.asm.mixin.extensibility.IRemapper;
+
+import net.fabricmc.mappingio.tree.MappingTree;
+
+public class MixinRemapper implements IRemapper {
+	protected final MappingTree mappings;
+	protected final int fromId;
+	protected final int toId;
+
+	public MixinRemapper(MappingTree mappings, int fromId, int toId) {
```

Comment:
It may be nice to move this to Mixin, and take an asm `Remapper`. MIO could then have an API to create a `Remapper` from a MappingTree.